Prompt to Download Immediately After Key Generation

George-cl opened the pull request:
Adds a prompt to download the key with a disclaimer that cancelling it is at the user's own risk.

The prompt means that a user is given an opportunity to back up at the point of creation and then is forced to backup at the point of deletion (although they can close the file explorer as described in the PR for that work).

![Screenshot from 2021-12-06 16-52-51](https://user-images.githubusercontent.com/69711689/144888287-13bd378a-7b7a-4ae9-b722-f4db05e1fe50.png)

### Asset
[Signer v1.4.5 (w/ bug fix)](https://github.com/casper-ecosystem/signer/files/7688656/casperlabs_signer-1.4.5.zip)

mssteuer commented:
I think there's an argument to be made to not allow account creation without saving the secret key. Currently pressing "Cancel" does nothing.
Also, while we're refactoring this, should we have the post-account-creation download be limited to JUST the secret key? The 3 files are confusing people.

mssteuer commented:
"If you choose not to, it is at your own risk and you may loose access to your funds in the future" - to make it as explicit as possible

mssteuer commented on src/popup/components/AccountPage.tsx:
```diff
@@ -89,41 +87,25 @@ class AccountPage extends React.Component<Props, State> {
+
+  downloadKeys(alias: string) {
+    confirm(
+      <div className="text-danger">Download Key</div>,
```
"Download Keys" (plural) or "Download Secret Key" (if we limit to just that?)

George-cl commented:
updated UI:
![image](https://user-images.githubusercontent.com/69711689/144925881-f40b700c-58ab-4d4d-b34f-589aa70166e3.png)

@mssteuer I've changed it so that they have to download the key to finish creating the account. With a small caveat that if the user closes the Signer before responding, the key will be stored in the Signer without it being backed up. Keys that haven't been backed up must be downloaded before they can be deleted.

I've also changed it so that downloading key~s~ only saves the `secret_key`

mssteuer commented:
> updated UI: ![image](https://user-images.githubusercontent.com/69711689/144925881-f40b700c-58ab-4d4d-b34f-589aa70166e3.png)
> 
> @mssteuer I've changed it so that they have to download the key to finish creating the account. With a small caveat that if the user closes the Signer before responding, the key will be stored in the Signer without it being backed up. Keys that haven't been backed up must be downloaded before they can be deleted.
> 
> I've also changed it so that downloading key~s~ only saves the `secret_key`

@George-cl awesome! I think that's a huge improvement! Final thoughts on the dialog text, which I think is still too "tech-y". My suggestion would be:

Caption: Download your key
Body: Before proceeding, download and securely save your secret key. Without it, you will not be able to recover access to your account.

I'm somewhat on the fence on the utility of the "Cancel" button, but I suppose it's best practice to have one, even though in this case it cancels you out of multiple steps (the previous screen being the "Create Account" screen presumably). Perhaps instead of "Cancel" and taking you home, should it be "Back" and take you to the "Create Account" screen? I could be convinced either way ;)

cspramit commented:
downloading key and re-importing the key gives as in screenshot with hardstop, this happens with a new and existing signe
![145_import](https://user-images.githubusercontent.com/80658431/144959298-be47f4bd-d84d-4093-95fc-0e3f671e6d22.PNG)
r

piotrwitek reviewed (approved):
Looks good

hoffmannjan reviewed (approved):
👍
